src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java

```diff
@@ -36,10 +37,38 @@ public class Shredder {
 
   private final DocumentLimitsConfig documentLimits;
 
+  private final JsonShreddingMetricsReporter jsonShreddingMetricsReporter;
+
   @Inject
-  public Shredder(ObjectMapper objectMapper, DocumentLimitsConfig documentLimits) {
+  public Shredder(
+      ObjectMapper objectMapper,
+      DocumentLimitsConfig documentLimits,
+      JsonShreddingMetricsReporter jsonShreddingMetricsReporter) {
     this.objectMapper = objectMapper;
     this.documentLimits = documentLimits;
+    this.jsonShreddingMetricsReporter = jsonShreddingMetricsReporter;
+  }
+
+  /**
+   * Wrapper function for shred function to add serialization metrics
+   *
+   * @param doc {@link JsonNode} to shred.
+   * @param commandName command name
+   * @return WritableShreddedDocument
+   */
+  public WritableShreddedDocument shredWithMetrics(JsonNode doc, String commandName) {
+    return shredWithMetrics(doc, null, commandName);
+  }
+
+  public WritableShreddedDocument shredWithMetrics(JsonNode doc, UUID txId, String commandName) {
+    // Start the timer
+    jsonShreddingMetricsReporter.startTimer();
```

Comment:
This is where we need to create reporter instance using metrics reporter factory object.